File: delivery_food_back_end/src/main/java/com/example/delivery_food_back_end/business/MenuItem.java
package com.example.delivery_food_back_end.business;

import java.io.Serializable;

public abstract class MenuItem implements Serializable
{
    static int id = 0;
    int idProduct;
    public abstract double computePrice();

    public abstract void setRating(double rating);

    public abstract double getCalories();

    public abstract void setCalories(double calories);

    public abstract double getProtein();

    public abstract void setProtein(double protein);

    public abstract double getFat();

    public abstract void setFat(double fat);

    public abstract double getSodium();

    public abstract void setSodium(double sodium);

    public abstract double getPrice();

    public abstract void setPrice(double price);

    public abstract String getTitle();

    public abstract void setTitle(String title);
}

File: delivery_food_front_end/lib/data/MenuItem.dart

class MenuItem {
   String title;
   double rating;
   double calories;
   double protein;
   double fat;
   double sodium;
   double price;

   MenuItem({required this.title,
    required this.rating,
    required this.calories,
    required this.protein,
    required this.fat,
    required this.sodium,
    required this.price});

  factory MenuItem.fromJson(Map<String, dynamic> json)
  {
    return MenuItem(title: json['title'],
        rating: json['rating'],
        calories: json['calories'],
        protein: json['protein'],
        fat: json['fat'],
        sodium: json['sodium'],
        price: json['price']);
  }

  Map<String,dynamic> toJson()=>
      {
        'title': this.title,
        'rating': this.rating,
        'calories': this.calories,
        'protein': this.protein,
        'fat': this.fat,
        'sodium': this.sodium,
        'price': this.price,

      };
}
